### DISCUSSION TITLE
Client-side routing

### DISCUSSION THREAD

**Discussion by TCBroad** [frandiox/vite-ssr#101] [Q&A] [answered]
Hi,

Thanks for the plugin - I've managed to convert some complex vue2 sites to vue3 using it but have a question regarding routing: 
Whenever I click an internal link on the site, it causes a full page load and executes on the server.
Is there an option to have the initial load run on the server but then subsequent page interactions run on the client? I haven't seen it mentioned in the documentation but could've missed something!

Cheers, 

Tom

**Comment by frandiox** [comment, marked as answer]
You need to use `vue-router`'s `RouterLink` component for that 👍

**Reply by TCBroad** [reply]
Hi,

I am using `RouterLink` anywhere a link is present but it still seems to do a full load. e.g.:

![image](https://user-images.githubusercontent.com/1906419/133766557-a18f23c5-d09d-4abf-8e05-2204eebaa9e3.png)

I'm using express for development with vite-ssr in middleware mode but just tried it the "normal" way and the behaviour is the same.

Nothing seems to run client-side other than the initial vite connection

![image](https://user-images.githubusercontent.com/1906419/133767672-b504524c-2aa0-43e1-aece-52a1b54b9f56.png)

**Reply by frandiox** [reply]
Hmm, that's weird. Can you make a small repro?
Or perhaps check any of the examples or vitesse-ssr template to see how things are done there.

The demo with vitesse-ssr is here: https://vitesse-ssr.vercel.app/
You can see there when clicking on the About link (second icon from the right) that it doesn't make a rendering request.

**Reply by TCBroad** [reply]
I've been chatting with another developer and I'm going to try running the built version to see if it's a problem in development only. If that's the case then I'm not too bothered about having it during dev. I'll have a look at the app you suggested and make sure I'm doing the same things. If I still can't get anywhere I'll try and reproduce it in a repo.

Thanks for the help!

**Reply by TCBroad** [reply]
So I started the project this morning and the SPA handover is now working as expected - even in dev mode! I didn't change anything so not sure what happened there but it does all seem to work properly. I can only assume I had something misconfigured and it was being cached across restarts until today. Who knows?!
